packages/teams-js/src/public/location.ts

```diff
@@ -46,82 +44,107 @@ export namespace location {
+      throw { errorCode: ErrorCode.INVALID_ARGUMENTS };
+    }
+
+    sendMessageToParent('location.getLocation', [props], callback);
+  }
+
   /**
    * @deprecated
-   * As of 2.0.0, please use {@link location.showLocation location.showLocation(location: Location): Promise\<void\>} instead.
+   * As of 2.0.0-beta.4, please use {@link location.map.showLocation location.map.showLocation(location: Location): Promise\<void\>} instead.
    * Shows the location on map corresponding to the given coordinates
    * @param location {@link Location} - which needs to be shown on map
    * @param callback - Callback to invoke when the location is opened on map
    */
-  export function showLocation(location: Location, callback: (error: SdkError, status: boolean) => void): void;
-  export function showLocation(
-    location: Location,
-    callback?: (error: SdkError, status: boolean) => void,
-  ): Promise<void> {
+  export function showLocation(location: Location, callback: (error: SdkError, status: boolean) => void): void {
+    if (!callback) {
+      throw new Error('[location.showLocation] Callback cannot be null');
+    }
     ensureInitialized(FrameContexts.content, FrameContexts.task);
-    return callCallbackWithErrorOrBooleanFromPromiseAndReturnPromise<void>(showLocationHelper, callback, location);
+    if (!isCurrentSDKVersionAtLeast(locationAPIsRequiredVersion)) {
+      throw { errorCode: ErrorCode.OLD_PLATFORM };
+    }
+    if (!location) {
+      throw { errorCode: ErrorCode.INVALID_ARGUMENTS };
+    }
+
+    sendMessageToParent('location.showLocation', [location], callback);
   }
 
-  export function showLocationHelper(location: Location): Promise<void> {
-    return new Promise<void>(resolve => {
-      if (!isCurrentSDKVersionAtLeast(locationAPIsRequiredVersion)) {
-        throw { errorCode: ErrorCode.OLD_PLATFORM };
-      }
+  export namespace map {
+    /**
+     * Allows user to choose location on map
+     * @returns The location chosen by the user after closing the map
+     */
+    export function chooseLocation(): Promise<Location> {
+      ensureInitialized(FrameContexts.content, FrameContexts.task);
+      return sendAndHandleError('location.getLocation', { allowChooseLocation: true, showMap: true });
+    }
+
+    /**
+     * Shows the location on map corresponding to the given coordinates
+     * @param location {@link Location} - which needs to be shown on map
+     * @returns Promise that resolves when the location dialog has been closed
+     */
+    export function showLocation(location: Location): Promise<void> {
+      ensureInitialized(FrameContexts.content, FrameContexts.task);
       if (!location) {
         throw { errorCode: ErrorCode.INVALID_ARGUMENTS };
       }
-      if (!isSupported()) {
-        throw errorNotSupportedOnPlatform;
-      }
-      resolve(sendAndHandleError('location.showLocation', location));
-    });
-  }
+      return sendAndHandleError('location.showLocation', location);
```

Comment:
If we make this `sendAndHandleError('location.getLocation', { allowChooseLocation: false, showMap: true})` it would help make sure that teams continues to work with the new functions as well!

Reply:
per our convo, I will be keeping it the way it is.  showLocation is separate function than getLocation
`showLocation` takes only location parameter and show the location on map. By changing it to the new way, 

We will be breaking the back compatibility because in v1 (and in v2), previously, we send location.showLocation to the hubSDK.

showLocation takes co-ordinators in and show them on a map. If we don't send the location coordinators, host won't have anything to show on map